### PR TITLE
Julia 1.0

### PR DESCRIPTION
**Pull request by MaximeBouton**
Some minor fixes to make it work on Julia 1.0

### CODE REVIEW

**Comment by MaximeBouton** [comment]
The build script is not working on 1.0 yet. `BinDeps` needs to be replaced by `BinaryProvider` and/or `BinaryBuilder`.

**Comment by visr** [comment]
If you'd like to give BinaryBuilder a try, I can help if needed. Doesn't seem like a library that is difficult to compile. You can base it off JuliaGeo/GEOSBuilder for instance.

**Comment by MaximeBouton** [comment]
I am working on this again, any suggestion on where to host the LibSpatialIndexBuilder repository? 
(I am happy to host it on my personal github but maybe you want it part of JuliaGeo?)

**Comment by yeesian** [comment]
Yeah, you should start them on your personal github :) It won't be hard to transfer ownership to JuliaGeo after, if that's something you want to do. Thanks for taking a crack at it!

**Comment by MaximeBouton** [comment]
https://github.com/MaximeBouton/LibSpatialIndexBuilder
First step, I could not get the wizard to push automatically to github so I need to set up travis manually...

**Comment by MaximeBouton** [comment]
I naively adapted the build script from GEOSBuilder: 
https://github.com/MaximeBouton/LibSpatialIndexBuilder/blob/master/.travis.yml

However I am missing the `secure` key, any idea on how to generate it? I created a token for the repository but then I am not sure what to do, the travis documentation is not easy to parse...

**Comment by visr** [comment]
Nice!

Perhaps you can use this function to create a token?
https://juliapackaging.github.io/BinaryBuilder.jl/latest/reference.html#BinaryBuilder.setup_travis-Tuple{Any}

The version here should be 1.8.5 I believe, since you are building libspatialindex 1.8.5: 
https://github.com/MaximeBouton/LibSpatialIndexBuilder/blob/61ba4aceb850d97858bbf0fda56db03bfdaabd66/build_tarballs.jl#L6

**Comment by MaximeBouton** [comment]
The `setup_travis` function failed, It is waiting forever here: 
https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/master/src/wizard/deploy.jl#L175

**Comment by MaximeBouton** [comment]
This was the issue: https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/265
I was giving the full URL instead of just the repo name.

**Comment by yeesian** [comment]
Thanks so much for this @MaximeBouton! I believe the script is now working here, and we just need to [support windows and darwin in LibSpatialIndexBuilder](https://github.com/MaximeBouton/LibSpatialIndexBuilder/pull/1) before this PR gets merged.

File: deps/build.jl [hunk 1]
```diff
@@ -1,50 +1,46 @@
...
+    Linux(:i686, libc=:glibc) => ("$bin_prefix/LibSpatialIndexBuilder.v1.8.5.i686-linux-gnu.tar.gz", "83ee51d110047dcea511c8eedf42430ce99094cb310dcb8b616dd952b90e838f"),
+    Linux(:i686, libc=:musl) => ("$bin_prefix/LibSpatialIndexBuilder.v1.8.5.i686-linux-musl.tar.gz", "e5aeca47bf2b0fd253faeee212f69e0cf174273eacd2e95b9c45a8a1b0962d2f"),
+    Linux(:powerpc64le, libc=:glibc) => ("$bin_prefix/LibSpatialIndexBuilder.v1.8.5.powerpc64le-linux-gnu.tar.gz", "4c5766af02db1028d08e87d8c19799b8d336cad247177083b4de1eee6ada24b9"),
+    Linux(:x86_64, libc=:glibc) => ("$bin_prefix/LibSpatialIndexBuilder.v1.8.5.x86_64-linux-gnu.tar.gz", "334dace9eabf6717d95befd3f781eb1492397c306ad424f5fdeaa75a559719e6"),
+    Linux(:x86_64, libc=:musl) => ("$bin_prefix/LibSpatialIndexBuilder.v1.8.5.x86_64-linux-musl.tar.gz", "4e392c2b391aa10eedb7e1e0034582f6cca0474fd9833a2966b860c99bfe6be0"),
+    FreeBSD(:x86_64) => ("$bin_prefix/LibSpatialIndexBuilder.v1.8.5.x86_64-unknown-freebsd11.1.tar.gz", "f7cab3a4f00dfefd5d6f55b4d390bcd37f46bfe5f030c23e92398f96fcdb58c6"),
+)
```
Comment (yeesian) [inline deps/build.jl:29]: we should update this list when we support windows and darwin in LibSpatialIndexBuilder

Comment (MaximeBouton) [reply deps/build.jl:29]: Yes I dropped support for Windows and MacOS just because I had no clue how to fix it. I will take a close look at your other PR.

**Comment by MaximeBouton** [comment]
Seems to work now. The appveyor build on windows 32bits was failing: 
https://ci.appveyor.com/project/NgYeeSian/libspatialindex-jl/build/job/evt46rfjcgn0ty2u

**Comment by yeesian** [comment]
Can you cherrypick the commit in https://github.com/yeesian/LibSpatialIndex.jl/commit/e5434d192bd77827072ccc5539ef3a95a41ac15c to fix the windows 32 error?

**Comment by MaximeBouton** [comment]
Should we enforce the arrays to be float32 in the tests? is that what is happening?
Here is a new error: 
```
Simple Tutorial: Error During Test at C:\projects\libspatialindex-jl\test\runtests.jl:5
  Got exception MethodError(Base.unsafe_convert, (Ptr{UInt64}, Base.RefValue{UInt32}(0x0d28bf34)), 0x00006c59) outside of a @test
  MethodError: no method matching unsafe_convert(::Type{Ptr{UInt64}}, ::Base.RefValue{UInt32})
  Closest candidates are:
    unsafe_convert(!Matched::Type{Ptr{Nothing}}, ::Base.RefValue{T}) where T at refvalue.jl:30
    unsafe_convert(::Type{T<:Ptr}, !Matched::T<:Ptr) where T<:Ptr at essentials.jl:352
    unsafe_convert(::Type{P<:Ptr}, !Matched::Ptr) where P<:Ptr at essentials.jl:353
    ...
  Stacktrace:
   [1] Index_Intersects_count(::Ptr{Nothing}, ::Array{Float64,1}, ::Array{Float64,1}, ::UInt32, ::Base.RefValue{UInt32}) at C:\projects\libspatialindex-jl\src\include\sidx_api.jl:82
   [2] macro expansion at C:\projects\libspatialindex-jl\test\runtests.jl:25 [inlined]
   [3] macro expansion at C:\cygwin\home\Administrator\buildbot\worker\package_win32\build\usr\share\julia\stdlib\v0.7\Test\src\Test.jl:1079 [inlined]
   [4] top-level scope at C:\projects\libspatialindex-jl\test\runtests.jl:7
   [5] include at .\boot.jl:317 [inlined]
   [6] include_relative(::Module, ::String) at .\loading.jl:1038
   [7] include(::Module, ::String) at .\sysimg.jl:29
   [8] include(::String) at .\client.jl:398
   [9] top-level scope at none:0
   [10] eval(::Module, ::Any) at .\boot.jl:319
   [11] exec_options(::Base.JLOptions) at .\logging.jl:317
   [12] _start() at .\client.jl:432
```

**Comment by yeesian** [comment]
I think [this line](https://github.com/yeesian/LibSpatialIndex.jl/blob/f21d89f6d49bcfca70f9386556007be356044951/test/runtests.jl#L22) needs to be changed to `nresults = Ref{UInt64}()`.

**Comment by MaximeBouton** [comment]
Seems like it work! Thanks @yeesian @visr

**Comment by yeesian** [comment]
Thanks so much for seeing this through to completion @MaximeBouton, I really appreciate it! :)

**Pull request merged**